lms/djangoapps/instructor/static/instructor/ProblemBrowser/data/actions/problemResponses.js

```diff
@@ -0,0 +1,84 @@
+/* global gettext */
+import { fetchTaskStatus, initiateProblemResponsesRequest } from '../api/client';
+import {
+  REPORT_GENERATION_ERROR,
+  REPORT_GENERATION_REQUEST,
+  REPORT_GENERATION_SUCCESS,
+  REPORT_GENERATION_TIMEOUT,
+} from './constants';
+
+const taskStatusSuccess = (succeeded, inProgress, reportPath, reportName) => ({
+  type: REPORT_GENERATION_SUCCESS,
+  succeeded,
+  inProgress,
+  reportPath,
+  reportName,
+});
+
+const problemResponsesRequest = blockId => ({
+  type: REPORT_GENERATION_REQUEST,
+  blockId,
+});
+
+const problemResponsesFailure = error => ({
+  type: REPORT_GENERATION_ERROR,
+  error,
+});
+
+const timeoutSet = timeout => ({
+  type: REPORT_GENERATION_TIMEOUT,
+  timeout,
+});
+
+const getTaskStatus = (endpoint, taskId) => dispatch =>
+  fetchTaskStatus(endpoint, taskId)
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error(response);
+    })
+    .then(
+      (statusData) => {
+        if (statusData.in_progress) {
+          const timeout = setTimeout(() => dispatch(getTaskStatus(endpoint, taskId)), 2000);
+          dispatch(timeoutSet(timeout));
+        }
+        return dispatch(
+          taskStatusSuccess(
+            statusData.task_state === 'SUCCESS',
+            statusData.in_progress,
+            statusData.task_progress.report_path,
```

Comment:
When I tested on the sandbox using `block-v1:edX+DemoX+Demo_Course+type@sequential+block@basic_questions` as the selected section, this line resulted in an error in the JS console:

```
Uncaught (in promise) TypeError: Cannot read property 'report_path' of undefined
    at ProblemBrowser.620abcae5227237e75d8.9c443a57bd0e.js:formatted:625
```
However, I can't reproduce it now. Perhaps some caching issue was involved. But please take a look in case you can see some sort of race condition or other code path that could trigger this. It still worked fine though.

Reply:
Will investigate.